### ISSUE TITLE
🐛 Bug Report: createFloatAttribute() 'toDouble' was called on null.

### ISSUE THREAD

**Issue by stonith404** [appwrite/sdk-for-dart#16]
### 👟 Reproduction steps

Execute the following function:

```
database.createFloatAttribute(
          collectionId: "collection",
          key: "someKey",
          xrequired: true)
```

### 👍 Expected behavior

The function should be executed without an error.

### 👎 Actual Behavior

The function returns:

```
NoSuchMethodError: The method 'toDouble' was called on null.
Receiver: null
Tried calling: toDouble()
#0      Object.noSuchMethod (dart:core-patch/object_patch.dart:68:5)
#1      new AttributeFloat.fromMap (package:dart_appwrite/src/models/attribute_float.dart:49:32)
#2      Database.createFloatAttribute (package:dart_appwrite/services/database.dart:279:34)
<asynchronous suspension>
```

As I can see the method `toDouble` is called on parameter that is not required. The problem is the `min`,`max` or `xdefault` parameter.

### 🎲 Appwrite version

Different version (specify in environment)

### 💻 Operating system

Linux

### 🧱 Your Environment

V0.12

### 👀 Have you spent some time to check if this issue has been raised before?

- [X] I checked and didn't find similar issue

### 🏢 Have you read the Code of Conduct?

- [X] I have read the [Code of Conduct](https://github.com/appwrite/appwrite/blob/HEAD/CODE_OF_CONDUCT.md)

**Comment by generalxhd** [comment]
@wess Moved.

**Comment by sbergmair** [comment]
Can confirm this behaviour
```dart
factory AttributeFloat.fromMap(Map<String, dynamic> map) {
    return AttributeFloat(
      key: map['key'].toString(),
      type: map['type'].toString(),
      status: map['status'].toString(),
      xrequired: map['required'],
      array: map['array'],
      min: map['min'].toDouble(),
      max: map['max'].toDouble(),
      xdefault: map['default'].toDouble(),
    );
  }
```
xdefault isn't required, therefore it can be null

**Comment by lohanidamodar** [comment]
Should be fixed on `dart_appwrite: 4.0.2`
